libcyphal_demo/CMakeLists.txt

```diff
@@ -11,6 +11,14 @@ project(libcyphal_demo
         HOMEPAGE_URL https://github.com/OpenCyphal-Garage/libcyphal)
 
 set(CMAKE_CXX_STANDARD "14" CACHE STRING "C++ standard to use when compiling.")
+set(DISABLE_CPP_EXCEPTIONS ON CACHE STRING "Disable C++ exceptions.")
+
+if (DISABLE_CPP_EXCEPTIONS)
+    message(STATUS "DISABLE_CPP_EXCEPTIONS is true. Adding -fno-exceptions to compiler flags.")
+    list(APPEND CXX_FLAG_SET "-fno-exceptions")
```

Comment:
Maybe we need a check if CMAKE_CXX_COMPILER_ID is GNU or Clang? Other compilers are unlikely to parse this option correctly.

https://github.com/Zubax/dyshlo/blob/c035e4701243ada935afcecb51ad1d7409c0f8f3/CMakeLists.txt#L28-L31